src/zod-schema/address-validation.ts

```diff
@@ -0,0 +1,68 @@
+import { z } from 'zod';
+
+enum estados {
+  Acre = 'Acre',
+  Alagoas = 'Alagoas',
+  Amapá = 'Amapá',
+  Amazonas = 'Amazonas',
+  Bahia = 'Bahia',
+  Ceará = 'Ceará',
+  DistritoFederal = 'Distrito Federal',
+  EspiritoSanto = 'Espírito Santo',
+  Goiás = 'Goiás',
+  Maranhão = 'Maranhão',
+  MatoGrosso = 'Mato Grosso',
+  MatoGrossoDoSul = 'Mato Grosso do Sul',
+  MinasGerais = 'Minas Gerais',
+  Pará = 'Pará',
+  Paraiba = 'Paraíba',
+  Paraná = 'Paraná',
+  Pernambuco = 'Pernambuco',
+  Piaui = 'Piauí',
+  RioDeJaneiro = 'Rio de Janeiro',
+  RioGrandeDoNorte = 'Rio Grande do Norte',
+  RioGrandeDoSul = 'Rio Grande do Sul',
+  Rondonia = 'Rondônia',
+  Roraima = 'Roraima',
+  SantaCatarina = 'Santa Catarina',
+  SaoPaulo = 'São Paulo',
+  Sergipe = 'Sergipe',
+  Tocantins = 'Tocantins',
+}
+
+export const AddressValidationSchema = z.object({
+  userId: z.number(),
+  cep: z.string().length(9, 'O cep deve conter 8 digitos e seguir o formato 00000-000'),
+  street: z
+    .string()
+    .min(10, 'O nome da rua deve conter pelo menos 10 caracteres')
+    .max(35, 'O nome da rua deve conter no máximo 35 caracteres'),
```

Comment:
pq esse máximo para nome da rua? não tem ruas com mais caracteres?

Reply:
Pelo que eu to vendo é algo que depende mais de regra de negócio mas por precaução e como eu vi algumas pessoas na internet fazendo eu vou deixar com 255 caracteres